src/check_names.cpp

```diff
@@ -0,0 +1,74 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// jsonify a character vector
+std::string jsonify(CharacterVector x) {
+  std::string out = "[";
+  for (int i = 0; i < x.size(); i++) {
+    out += "\"" + x[i] + "\",";
+  }
+  // trim trailing comma
+  if (x.size() > 0) {
+    out = out.substr(0, out.size() - 1);
```

Comment:
This copies the entire string. You could just do out.resize(out.size() - 1) instead which should be much more efficient.
Alternatively, you can append the first element and then loop over the remaining elements appending the separator first.